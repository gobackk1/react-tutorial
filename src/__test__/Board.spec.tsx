import React from "react";
import Board from "../components/Board";
import { Mark } from "../types";
import { mount, ReactWrapper } from "enzyme";
import { mountToJson } from "enzyme-to-json";

describe("Boardコンポーネントのテスト", () => {
  const mockSquares: Mark[] = [
    "X",
    "O",
    null,
    null,
    null,
    null,
    null,
    null,
    null
  ];
  const handleClickSpy = jest.fn();
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(
      <Board
        squares={mockSquares}
        onClick={() => {
          handleClickSpy();
        }}
      />
    );
  });

  describe("propsのテスト", () => {
    test("squaresが正しく表示されていること", () => {
      expect(wrapper.find('[data-index="0"]').text()).toContain("X");
      expect(wrapper.find('[data-index="1"]').text()).toContain("O");
      for (let i = 2; i < mockSquares.length; i++) {
        expect(wrapper.find(`[data-index="${i}"]`).text()).toContain("");
      }
    });

    test("onClickがSquareへ渡されていること", () => {
      const square = wrapper.find('[data-index="0"]');
      square.simulate("click");
      expect(handleClickSpy).toHaveBeenCalled();
    });
  });

  test("snapshot", () => {
    const snapshot = mountToJson(wrapper);
    expect(snapshot).toMatchSnapshot();
  });
});
