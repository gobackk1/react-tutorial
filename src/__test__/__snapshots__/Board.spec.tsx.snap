// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boardコンポーネントのテスト snapshot 1`] = `
<Board
  onClick={[Function]}
  squares={
    Array [
      "X",
      "O",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ]
  }
>
  <div>
    <div
      className="board-row"
    >
      <Square
        data-index="0"
        onClick={[Function]}
        value="X"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
      </Square>
      <Square
        data-index="1"
        onClick={[Function]}
        value="O"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          O
        </button>
      </Square>
      <Square
        data-index="2"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
    </div>
    <div
      className="board-row"
    >
      <Square
        data-index="3"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
      <Square
        data-index="4"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
      <Square
        data-index="5"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
    </div>
    <div
      className="board-row"
    >
      <Square
        data-index="6"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
      <Square
        data-index="7"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
      <Square
        data-index="8"
        onClick={[Function]}
        value={null}
      >
        <button
          className="square"
          onClick={[Function]}
        />
      </Square>
    </div>
  </div>
</Board>
`;
